src/TcpAdapterProxy.cpp

```diff
@@ -1101,7 +1106,7 @@ namespace aws { namespace iot { namespace securedtunneling {
                             BOOST_LOG_SEV(log, error) << (boost::format("Could not listen on bind address: %1% -- %2%")
                                 % results->endpoint().address().to_string() % listen_ec.message()).str();
                             basic_retry_execute(log, retry_config,
-                                [&tac, &ec]() { throw proxy_exception((boost::format("Failed to listen on bind address %1%:%2%") % tac.bind_address_actual % tac.adapter_config.data_port).str(), ec); });
+                                [&tac, &listen_ec]() { throw proxy_exception((boost::format("Failed to listen on bind address %1%:%2%") % tac.bind_address_actual % tac.adapter_config.data_port).str(), listen_ec, true); });
```

Comment:
I found that we were incorrectly reporting a "success" error code due to referencing the wrong error code variable here.